src/components/course/data/utils.jsx

```diff
@@ -619,29 +631,101 @@ export const getMissingSubsidyReasonActions = ({
-        as={Hyperlink}
-        destination={`mailto:${adminEmails}`}
+        as={MailtoLink}
+        className="text-center"
+        to={adminEmails}
         target="_blank"
         size="sm"
         block
       >
         Contact administrator
-        <span className="sr-only">about funds</span>
+        <span className="sr-only">for help</span>
       </Button>
     );
   }
 
   return null;
 };
 
+export const getSubscriptionDisabledEnrollmentReasonType = ({
+  customerAgreementConfig,
+  catalogsWithCourse,
+  subscriptionLicense,
+  hasEnterpriseAdminUsers,
+}) => {
+  const subscriptionsApplicableToCourse = customerAgreementConfig?.subscriptions?.filter(
+    subscription => catalogsWithCourse.includes(subscription?.enterpriseCatalogUuid),
+  );
+  const hasExpiredSubscriptions = subscriptionsApplicableToCourse?.filter(
+    subscription => subscription.daysUntilExpirationIncludingRenewals < 0,
+  )?.length > 0;
+  const hasExhaustedSubscriptions = subscriptionsApplicableToCourse?.filter(
+    subscription => subscription?.licenses?.unassigned === 0,
+  )?.length > 0;
+  const applicableSubscriptionNonExpiredNonExhausted = subscriptionsApplicableToCourse?.find(
+    subscription => subscription.daysUntilExpirationIncludingRenewals >= 0 && subscription?.licenses?.unassigned > 0,
+  );
+
+  if (hasExpiredSubscriptions) {
+    // If customer's subscription plan(s) containing the course being viewed have expired,
+    // change `reasonType` to use the `SUBSCRIPTION_EXPIRED` message.
+    if (hasEnterpriseAdminUsers) {
+      return DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_EXPIRED;
+    }
+    return DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_EXPIRED_NO_ADMINS;
+  }
+
+  if (hasExhaustedSubscriptions) {
+    // If customer's subscription plan(s) containing the course being viewed no longer have
+    // any remaining seats, change `reasonType` to use the `SUBSCRIPTION_SEATS_EXHAUSTED` message.
+    if (hasEnterpriseAdminUsers) {
+      return DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_SEATS_EXHAUSTED;
+    }
+    return DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_SEATS_EXHAUSTED_NO_ADMINS;
+  }
+
+  if (subscriptionLicense?.status === LICENSE_STATUS.REVOKED) {
+    // If learner's subscription license is revoked, change `reasonType` to use
+    // the `SUBSCRIPTION_DEACTIVATED` message.
+    return DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_DEACTIVATED;
+  }
+
+  if (applicableSubscriptionNonExpiredNonExhausted) {
+    // If customer has a subscription plan(s) containing the course being viewed that is not expired
+    // nor exhausted, change `reasonType` to use the `SUBSCRIPTION_LICENSE_NOT_ASSIGNED` message.
+    if (hasEnterpriseAdminUsers) {
+      return DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_LICENSE_NOT_ASSIGNED;
+    }
+    return DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_LICENSE_NOT_ASSIGNED_NO_ADMINS;
+  }
+
+  // There is no applicable subscriptions-related reason for disabled enrollment.
+  return undefined;
```

Comment:
Since the structure of the conditional is very similar across a majority of cases, it looks like we can extract it out based on `hasEnterpriseAdminUsers`

```js
function returnReasonType({
  hasExpiredSubscriptions,
  hasExhaustedSubscriptions,
  applicableSubscriptionNonExpiredNonExhausted,
  hasEnterpriseAdminUsers,
}, returnMessage) {
  if ((hasExpiredSubscriptions || hasExhaustedSubscriptions || applicableSubscriptionNonExpiredNonExhausted) && hasEnterpriseAdminUsers) {
    return returnMessage.hasAdmins;
  }
  return returnMessage.noAdmins;
}
```

Then it could be refactored as below

```js
return returnReasonType({
  hasExpiredSubscriptions,
  hasEnterpriseAdminUsers,
}, {
  hasAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_EXPIRED,
  noAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_EXPIRED_NO_ADMINS,
});
return returnReasonType({
  hasExhaustedSubscriptions,
  hasEnterpriseAdminUsers,
}, {
  hasAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_SEATS_EXHAUSTED,
  noAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_SEATS_EXHAUSTED_NO_ADMINS,
});

return returnReasonType({
  applicableSubscriptionNonExpiredNonExhausted,
  hasEnterpriseAdminUsers,
}, {
  hasAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_LICENSE_NOT_ASSIGNED,
  noAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_LICENSE_NOT_ASSIGNED_NO_ADMINS,
});

  if (subscriptionLicense?.status === LICENSE_STATUS.REVOKED) {
    // If learner's subscription license is revoked, change `reasonType` to use
    // the `SUBSCRIPTION_DEACTIVATED` message.
    return DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_DEACTIVATED;
  }

  return undefined;
```